### ISSUE TITLE
Carthage 0.20 exits early if a dependency has no shared framework schemes

### ISSUE THREAD

**Issue by dcaunt** [Carthage/Carthage#1792]
* `carthage version`: 0.20.0
* `xcodebuild -version`: 8.2 (8C38)
* Are you using `--no-build`? no
* Are you using `--no-use-binaries`? no
* Are you using `--use-submodules`? no

**Cartfile**
```
github "AFNetworking/AFNetworking" == 2.6.3
github "robb/Cartography" ~> 1.0
```

**Carthage Output (0.20)**

```
$ carthage bootstrap --platform iOS
*** Checking out AFNetworking at "2.6.3"
*** Checking out Cartography at "1.0.1"
*** xcodebuild output can be found in /var/folders/t6/566y64zd1c54v7tkfq9jx4300000gn/T/carthage-xcodebuild.OduibX.log
Dependency "AFNetworking" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/AFNetworking/AFNetworking/issues/new
```

**Carthage Output (0.19.1)**

```
$ carthage bootstrap --platform iOS
*** Please update to the latest Carthage version: 0.20.0. You currently are on 0.19.1
*** No Cartfile.resolved found, updating dependencies
*** Fetching AFNetworking
*** Fetching Cartography
*** Checking out AFNetworking at "2.6.3"
*** Checking out Cartography at "1.0.1"
*** xcodebuild output can be found in /var/folders/t6/566y64zd1c54v7tkfq9jx4300000gn/T/carthage-xcodebuild.fINtq1.log
*** Skipped building AFNetworking due to the error:
Dependency "AFNetworking" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/AFNetworking/AFNetworking/issues/new
*** Building scheme "Cartography-iOS" in Cartography.xcworkspace
```

Looks like a regression from 0.19.1 to 0.20.

**Comment by mdiep** [comment]
😩

**Comment by mdiep** [comment]
If someone wants to fix this, a test would be very much appreciated!

**Comment by gperdomor** [comment]
:(

**Comment by mdiep** [comment]
I started looking at this today. It's going to require some refactoring to fix up how projects are built. I'll try to fix it up.

**Comment by tomekc** [comment]
Strange, I am having the same issue, trying to build AFNetworking too. It is supposed to be fixed back then in 0.20.x but I have everything up-to date:

Carthage: 0.23.0
Xcode: 8.3.3

Cartfile:

```
github "xmartlabs/XLForm" == 3.3.0
github "AFNetworking/AFNetworking" == 2.6.3
```

When invoked

`$ carthage update --platform ios --cache-builds`

I am getting:

```
*** Fetching XLForm
*** Fetching AFNetworking
*** Checking out XLForm at "3.3.0"
*** Checking out AFNetworking at "2.6.3"
*** xcodebuild output can be found in /var/folders/x_/hlz9lbfd30l172slh8jxk9ww0000gn/T/carthage-xcodebuild.vc6B35.log
*** No cache found for AFNetworking, building with all downstream dependencies
*** Valid cache found for XLForm, skipping build
*** Skipped building AFNetworking due to the error:
Dependency "AFNetworking" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/AFNetworking/AFNetworking/issues/new
```

**Comment by BobElDevil** [comment]
@tomekc This isn't a carthage bug, the error message is correct in this case. Carthage support wasn't added to AFNeworking until 3.0.0. See
AFNetworking/AFNetworking#3121
